packages/react-dom-bindings/src/client/ReactDOMComponent.js

```diff
@@ -1099,6 +1099,10 @@ export function setInitialProperties(
           }
         }
       }
+      // Fix #27858: input[type="file"] has cancel event
+      if (type === 'file') {
+        listenToNonDelegatedEvent('cancel', domElement);
+      }
```

Comment:
Can we move the other listenToNonDelegatedEvent call in this branch to be next to this?

Also no need for the comments in this file; they don't really add info beyond what is evident from the code itself.

Reply:
done❤